load-testing/tests/relays_stress_helpers_test.go

```diff
@@ -0,0 +1,1143 @@
+	loadTestManifestContent, err := os.ReadFile(loadTestManifestPath)
+	require.NoError(s, err)
+
+	provisionedActors, err := config.ParseLoadTestManifest(loadTestManifestContent)
+	require.NoError(s, err)
+
+	for _, gateway := range provisionedActors.Gateways {
+		s.gatewayUrls[gateway.KeyName] = gateway.ExposedUrl
+	}
+
+	for _, supplier := range provisionedActors.Suppliers {
+		s.suppliersUrls[supplier.KeyName] = supplier.ExposedUrl
+	}
+}
+
+// TODO_IN_THIS_COMMIT: godoc comment.
+func (s *relaysSuite) mapSessionInfoFn(
+	batchInfoPublishCh chan<- *batchInfoNotif,
+) channel.MapFn[client.Block, *sessionInfoNotif] {
+	var (
+		// The test suite is initially waiting for the next session to start.
+		waitingForFirstSession = true
+		prevBatchTime          time.Time
+	)
+
+	return func(
+		ctx context.Context,
+		block client.Block,
+	) (*sessionInfoNotif, bool) {
+		blockHeight := block.Height()
+		if blockHeight <= s.latestBlock.Height() {
+			return nil, true
+		}
+
+		sessionInfo := &sessionInfoNotif{
+			blockHeight:             blockHeight,
+			sessionNumber:           keeper.GetSessionNumber(blockHeight),
+			sessionStartBlockHeight: keeper.GetSessionStartBlockHeight(blockHeight),
+			sessionEndBlockHeight:   keeper.GetSessionEndBlockHeight(blockHeight),
+		}
+
+		infoLogger := logger.Info().
+			Int64("session_num", sessionInfo.sessionNumber).
+			Int64("block_height", block.Height())
+
+		// If the test has not started and the current block is not the first block
+		// of the session, wait for the next session to start.
+		if waitingForFirstSession && blockHeight != sessionInfo.sessionStartBlockHeight {
+			countDownToTestStart := sessionInfo.sessionEndBlockHeight - blockHeight + 1
+			infoLogger.Msgf(
+				"waiting for next session to start: in %d blocks",
+				countDownToTestStart,
+			)
+
+			// The test is not to be started yet, skip the notification to the downstream
+			// observables until the first block of the next session is reached.
+			return nil, true
+		}
+
+		// If the test has not started, set the start block height to the current block height.
+		// As soon as the test start, s.startBlockHeight will no longer be updated.
+		// It is updated only once at the start of the test.
+		if waitingForFirstSession {
+			s.startBlockHeight = blockHeight
+		}
+
+		// Mark the test as started.
+		waitingForFirstSession = false
+
+		// Log the test progress.
+		infoLogger.Msgf(
+			"test progress blocks: %d/%d",
+			blockHeight-s.startBlockHeight, s.testDurationBlocks,
+		)
+
+		// If the test duration is reached, cancel the context to stop the test.
+		if blockHeight >= s.startBlockHeight+s.testDurationBlocks {
+			logger.Info().Msg("Test done, cancelling scenario context")
+			s.cancelCtx()
+
+			return nil, true
+		}
+
+		// If the current block is the start of any new session, activate the prepared
+		// actors to be used in the current session.
+		s.activatePreparedActors(sessionInfo)
```

Comment:
Big fan of this function name. 

![activate-dino-fury-gold-ranger](https://github.com/pokt-network/poktroll/assets/1892194/64d2245a-8f58-4d17-b12b-ee0ac903bb4e)

Reply:
https://twitter.com/olshansky/status/1785102713078063476